ShopApp/Data/API/Alamofire/MagentoAPI.swift

```diff
@@ -154,14 +157,76 @@ public class MagentoAPI: BaseAPI, API {
     
     // MARK: - Categories
     
-    public func getCategoryList(perPage: Int, paginationValue: Any?, sortBy: SortingValue?, reverse: Bool, callback: @escaping RepoCallback<[ShopApp_Gateway.Category]>) {
-        // TODO: Implement api method
-        callback([], nil)
+    public func getCategoryList(perPage: Int, paginationValue: Any?, sortBy: SortingValue?, reverse: Bool, rootCategoryId: String?, callback: @escaping RepoCallback<[ShopApp_Gateway.Category]>) {
```

Comment:
Can we avoid passing `rootCategoryId` for each request? I think we can pass it to `MagentoApi` class